Notifications: do not rely on admin-bar styles

jeherve opened the pull request:
Follow up from #13450

#### Changes proposed in this Pull Request:

* In some cases, admin-bar styles may be dequeued (like when the Masterbar module is active). Let's consequently not rely on them to be there.

cc @westonruter 

#### Testing instructions:

* Before to apply this patch, enable both Notifications and the Masterbar module.
* Notice the broken layout.
* Apply patch.
* Notice that both the Masterbar and the Notifications work well.

#### Proposed changelog entry for your changes:

* None

jeherve committed:
Notifications: do not rely on admin-bar styles

Follow up from #13450

In some cases, admin-bar styles may be dequeued (like when the Masterbar module is active).

westonruter commented:
@jeherve If the stylesheet is not marked as having an `admin-bar` dependency, then you'll need to manually mark it as being part of AMP dev mode, as can be seen with the `script_loader_tag` filter for scripts. So something like this:

```php
if ( class_exists( 'Jetpack_AMP_Support' ) && Jetpack_AMP_Support::is_amp_request() ) {
	add_filter(
		'style_loader_tag',
		function ( $tag, $handle ) {
			if ( in_array( $handle, [ 'wpcom-notes-admin-bar', 'noticons' ], true ) ) {
				$tag = preg_replace( '/(?<=<link)(?=\s|>)/i', ' data-ampdevmode', $tag );
			}
			return $tag;
		},
		10,
		2
	);
}
```

jeherve committed:
Stop dequeuing style when we need it (when notes is active)

This also adds the dependency back:
https://github.com/Automattic/jetpack/pull/13690#issuecomment-539532230

kraftbj commented:
I'm marking this as approved since it does resolve the original issue.

When using AMP in canonical mode, the traditional admin bar is on the front end -- is this expected? No issues or errors, but different behavior with AMP on vs off.

kristarella commented:
> When using AMP in canonical mode, the traditional admin bar is on the front end -- is this expected? No issues or errors, but different behavior with AMP on vs off.

I noticed this when testing AMP on the weekend. Does Jetpack influence this? There used to be an option in the AMP plugin whether or not to display the admin bar, but that option is no longer there. IMO it would be better not to display the admin bar because it is adding extra CSS to the page that might affect the CSS treeshake (like bumping out other CSS for being over the 50kb limit).

westonruter commented:
As of AMP v1.3 the admin bar no longer counts against the 50KB limit. This is why there is no longer an option to disable the admin bar on AMP pages.

kraftbj commented:
To clarify, my curiosity was more that the traditional admin bar was present on an AMP page and not the wpcom masterbar. Between the options of the traditional admin bar or no admin bar at all, I'd like to see the traditional admin bar given AMP canonical would be the primary post view experience.

jeherve commented:
> since wpcom-notes-admin-bar has admin-bar as a dependency, would wp_dequeue_style() even do anything since it isn't being directly enqueued to begin with?

It does, yes. In practice, you end up with the admin bar not being loaded at all:
![image](https://user-images.githubusercontent.com/426388/66471824-51c9a000-ea8c-11e9-8df2-20af02ee2ce8.png)

> When using AMP in canonical mode, the traditional admin bar is on the front end -- is this expected

I'd say ye. As you point out the traditional admin bar seems like a better fit there.
